modules/markup/markup.go

```diff
@@ -62,11 +62,10 @@ func Type(filename string) string {
 // IsReadmeFile reports whether name looks like a README file
 // based on its name.
 func IsReadmeFile(name string) bool {
-	name = strings.ToLower(name)
 	if len(name) < 6 {
 		return false
-	} else if len(name) == 6 {
-		return name == "readme"
 	}
-	return name[:7] == "readme."
+
+	name = strings.ToLower(name)
+	return (name == "readme" || name[:7] == "readme.")
```

Comment:
That what I said in my second comment, this would also validate file like README_LANGUAGE.md.